dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+
+import WebCacheApp.CachedServlet;
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class WebCacheTest {
+
+    public static final String SERVER_NAME = "webCacheServer";
+    public static final String APP_NAME = "webCacheApp";
+    public static final long CACHE_MISS_THRESHHOLD = 150;
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        WebArchive webappWar = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addClass(CachedServlet.class)
+                        .addAsWebInfResource(new File(server.getInstallRoot() + "/usr/servers/" + server.getServerName() + "/cachespec.xml"));
+        ShrinkHelper.exportAppToServer(server, webappWar, DeployOptions.OVERWRITE);
+    }
+
+    private boolean isCacheHit(String endpoint, String response) throws IOException {
+        long startTime = System.currentTimeMillis();
+        HttpUtils.findStringInUrl(server, APP_NAME + endpoint, response);
+        long endTime = System.currentTimeMillis();
+        return (endTime - startTime) < CACHE_MISS_THRESHHOLD;
+    }
+
+    @Test
+    public void testWebCacheServlet() throws Exception {
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS);
+        server.startServer();
+        server.stopServer();
```

Comment:
calling `stopServer()` followed by `checkpointRestore()` should be needed with the simple setCheckpoint call above which will auto restore on `startServer`